src/lib/RangeHelper.js

```diff
@@ -386,6 +386,127 @@ define(function (require) {
 		};
 
 		/**
+		 * Splits the HTML tree until commonParent closing all tags and
+		 * reopening them after the split point.
+		 *
+		 * After calling this function, cutElement will be child of
+		 * commonParent, nextSibling and previousSibling of the element
+		 * which is the topmost parent of cutElement while still being
+		 * descendant of commonParent.
+		 *
+		 * Example (examples show result of the .outerHTML parameter applied
+		 * to "<strong>"):
+		 * If the commonParent is "<strong>" and cutElement is
+		 * "<!--break-->" in:
+		 * <strong>hi there, how <em>are <span>y<!--break-->ou</span>
+		 * doing</em> today</strong>
+		 * After executing:
+		 * <strong>hi there, how <em>are <span>y</span></em><!--break-->
+		 * <em><span>ou</span> doing</em> today?</strong>
+		 *
+		 *
+		 * @param {DOMElement} commonParent The common ancestor at clipping
+		 * @param {DOMNode} cutElement Where the closing and re-opening happens
+		 * @return undefined
+		 * @function
+		 * @name splitElement
+		 * @memberOf RangeHelper.prototype
+		 * @author http://stackoverflow.com/a/27498178/551625
+		 * @author brunoais
+		 */
+		base.splitElement = function (commonParent, cutElement) {
+			var grandparent;
+			for (var parent = cutElement.parentNode; commonParent !==
+					parent; parent = grandparent) {
+				var right = parent.cloneNode(false);
+
+				while (cutElement.nextSibling) {
+					right.appendChild(cutElement.nextSibling);
+				}
+
+				grandparent = parent.parentNode;
+				grandparent.insertBefore(right, parent.nextSibling);
+				grandparent.insertBefore(cutElement, right);
+			}
+		};
+
+		/**
+		 * Replace elements with provided selector inside user selection with
+		 * their children
+		 *
+		 * This method provides an easier way to have the same final aspect
+		 * for HTML as powerful as the browser's native execCommand() method
+		 * when removing HTML tag formatting.
+		 *
+		 * This method is mostly meant to be used with <b>inline</b> HTML
+		 * Elements. Having it working with block level elements is not
+		 * guaranteed.
+		 *
+		 *
+		 * @param {String} removeElementSelector Selector match what to delete
+		 * @return undefined
+		 * @function
+		 * @name splitAtSelection
+		 * @memberOf RangeHelper.prototype
+		 * @author brunoais
+		 */
+		base.splitAtSelection = function (removeElementSelector) {
+
+			var commonParent = base.getFirstBlockParent();
+
+			base.insertMarkers();
+
+			var startMarkerElement = base.getMarker(startMarker);
+			var endMarkerElement = base.getMarker(endMarker);
+			startMarkerElement.parentNode.normalize();
```

Comment:
To reproduce, Use multi-level combination of inline elements then do a collapsed selection followed by calling this method with an empty selector. Don't forget to add a return before the `.removeMarkers()` (later replaced by `.restoreSelection()`) so that you can see where the markers are.